packages/patterns/src/components/pattern-overrides-controls.js

```diff
@@ -34,12 +35,17 @@ function removeBindings( bindings, syncedAttributes ) {
 }
 
 function addBindings( bindings, syncedAttributes ) {
-	const updatedBindings = { ...bindings };
+	const updatedBindings = {
+		...bindings,
+		__default: { source: PATTERN_OVERRIDES_BINDING_SOURCE },
+	};
+	// Back-compat: Remove existing duplicate bindings.
 	for ( const attributeName of syncedAttributes ) {
-		if ( ! bindings?.[ attributeName ] ) {
-			updatedBindings[ attributeName ] = {
-				source: PATTERN_OVERRIDES_BINDING_SOURCE,
-			};
+		if (
+			updatedBindings[ attributeName ]?.source ===
+			PATTERN_OVERRIDES_BINDING_SOURCE
+		) {
+			delete updatedBindings[ attributeName ];
```

Comment:
There's also a migration here for the old 'pattern_overrides' source name for pattern overrides:
https://github.com/WordPress/gutenberg/blob/5727784de928e67ec07761cc92060ac08b07bc83/packages/blocks/src/api/parser/convert-legacy-block.js#L82-L111

This code is triggered during the block parsing phase, so it might be a better option to update that rather than triggering the migration on a user action. What do you think?

We can probably also consider dropping the migration from the old 'pattern_overrides' source name at the same time, it's been a few gutenberg versions now.